spec/diagnose_spec.rb

```diff
@@ -1283,28 +1287,51 @@
   end
 
   it "prints agent diagnose section with errors" do
+    expected_output = if @runner.type == :python
+                        [
+                          /Agent diagnostics/,
+                          /  Extension tests/,
+                          /  Configuration: invalid/,
+                          /     Error: RequiredEnvVarNotPresent\("APPSIGNAL_PUSH_API_KEY"\)/,
+                          /  Agent tests/,
```

Comment:
How come the Python behaviour is so different here? I guess the other integrations don't even start the agent when there's no push API key, right?

Maybe we'd want to replicate that behaviour in the Python integration?

Reply:
I'm not sure about how to do it. The extension acts like a middleware for the other integrations and that helps a lot.

Reply:
I think this is resolved now that we removed the extension section.